src/Symfony/Component/DependencyInjection/ContainerAwareInterface.php

```diff
@@ -15,6 +15,8 @@
  * ContainerAwareInterface should be implemented by classes that depends on a Container.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @deprecated since Symfony 6.4, use dependency injection instead
```

Comment:
let's trigger a deprecation notice here also and adjust the code base as needed

Reply:
ok, we won't be able to trigger at runtime because BundleInterface extends ContainerAwareInterface
let's still see where this is used and adjust

Reply:
`BundleInterface` is the last place where the interface is implemented in the codebase (removing it seems to keep tests green, good point for 7.0). I added a few lines to `Bundle` to get rid of `ContainerAwareTrait`